### PR TITLE
add initial draft of functions 

### PR DESCRIPTION
**Pull request by KevinMarquette**
add `Get-NpmPackageInfo` function 
add `Publish-NpmPackage` function 
add `Test-NpmPackage` function 
add tests and test data

### CODE REVIEW

File: NpmPS/Public/Publish-NpmPackage.ps1 [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+        $Tag,
+
+        # Force publish even if package already exists
+        [Switch]
+        $Force
+    )
+
+    begin
+    {
+        $npmProtocolRegex = '^http?:'
```
Comment (EklipZgit) [inline NpmPS/Public/Publish-NpmPackage.ps1]: should be '^https?' correct? optional s, not optional p?

File: Tests/data/npmpackage.json [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    "0.0.0-rc.5": "2018-11-01T02:02:41.59Z",
+    "0.0.1": "2018-11-01T02:02:41.59Z"
+  },
+  "versions": {
+    "0.0.0-rc.5": {
+      "description": "Contoso Component",
+      "main": "contoso-component.bundle.js",
+      "author": {
+        "name": "Collin Stevens",
+        "email": "costevens@loandepot.com"
```
Comment (EklipZgit) [inline Tests/data/npmpackage.json]: you sure you want this in here?

**Review by EklipZgit — CHANGES_REQUESTED** [review]

File: NpmPS/Public/Publish-NpmPackage.ps1 (hunk 1 above)
Comment (KevinMarquette) [reply NpmPS/Public/Publish-NpmPackage.ps1]: corrected

File: Tests/data/npmpackage.json (hunk 2 above)
Comment (KevinMarquette) [reply Tests/data/npmpackage.json]: i picked a different random username

**Review by EklipZgit — APPROVED** [review]

**Pull request merged**